Honor pool connection timeout when executing queries directly in the pool

pablosaavedra-rappi opened the pull request:
Should fix #1232, as it now uses the timeout when acquiring the connection

Motivation:

We've been bugged by #1232 for some time and the fix keeps getting pushed.

pablosaavedra-rappi commented:
Didn't find any test referencing PoolImpl that I could use to replicate the behavior

vietj commented:
I think the change is good but will impact pooled clients for which the promise is to schedule commands without waiting for a connection to be free, so we should have this behaviour configurable and disabled when producing a pooled client. See `PgBuilder#client()` vs `PgBuilder#pool()`

vietj reviewed (changes requested):
I think the change is good but will impact pooled clients for which the promise is to schedule commands without waiting for a connection to be free, so we should have this behaviour configurable and disabled when producing a pooled client. See `PgBuilder#client()` vs `PgBuilder#pool()`

pablosaavedra-rappi commented:
Thanks for the comment, will look into it and update the PR.

pablosaavedra-rappi commented:
@vietj finally got around to working on this one. I rebased with the 4.x branch and added the changes you requested. Let me know if you have any other comment.

Thanks.

vietj commented:
@pablosaavedra-rappi yes you can push generated source code

vietj commented:
Looking at this again, I am not a fan of the option as is, I think instead command should have their own timeout to explicitely named the feature, and we should call it `scheduleTimeout` which is the timeout to schedule a command, if a connection has not been made available within that timeout then the command cannot be scheduled and is failed

vietj commented:
@pablosaavedra-rappi

pablosaavedra-rappi commented:
So @vietj you are suggesting we overload the method with an additional `scheduleTimeout` parameter and the behavior there? I can check how that'd look and update the PR.

vietj commented:
instead of having a boolean that says that the timeout applies to
everything, we have a timeout that applies to command execution scheduling.

On Fri, May 17, 2024 at 4:39 PM pablosaavedra-rappi <
***@***.***> wrote:

> So @vietj <https://github.com/vietj> you are suggesting we overload the
> method with an additional scheduleTimeout parameter and the behavior
> there? I can check how that'd look and update the PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse-vertx/vertx-sql-client/pull/1338#issuecomment-2117754822>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABXDCUNBDJ2FYM2AIJSMP3ZCYJAVAVCNFSM6AAAAAA2GR6U62VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMJXG42TIOBSGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
